### PR TITLE
Various BuildTransportPackage.ps1 Refactors and Fixes

### PR DESCRIPTION
**Pull request by kythant**
Added LastExitCode checks in various parts of the BuildTransportPackage.ps1 script. 

Move Microsoft.Telemetry.Inbox.nupkg restore back to yml so that it can work with Nuget Authenticate on the feed.

Rename BuildTransportPackage.ps1 to BuildAll.ps1. Delete deprecated Build_All.cmd to reduce confusion.

### CODE REVIEW

File: BuildAll.ps1 [hunk 1]
```diff
@@ -154,12 +165,17 @@ Try {
         #------------------
 
         #Restore packages from mrt.
-        & .\.nuget\nuget.exe restore "$MRTSourcesDirectory\mrt\MrtCore.sln" -ConfigFile nuget.config
+        & .\.nuget\nuget.exe restore "$MRTSourcesDirectory\mrt\MrtCore.sln" -ConfigFile nuget.config -ConfigFile nuget.config -PackagesDirectory "$MRTSourcesDirectory\mrt\packages"
```
Comment (alexlamtest) [inline BuildAll.ps1:168]: It is unclear why "-ConfigFile nuget.config" is duplicated.

Comment (kythant) [reply BuildAll.ps1:168]: Whoops typo

**Review by alexlamtest — APPROVED** [review]

**Pull request merged**